### PR TITLE
Ensure that fields are respected in get_theme_item

### PR DESCRIPTION
**Pull request by spacedmonkey**
Trac ticket: https://core.trac.wordpress.org/ticket/54595

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by peterwilsoncc** [comment]
The test `Tests_Rewrite::test_url_to_postid_custom_post_type` failed in PHP 7.2 only, so I have re-run them to see if it was a random error/time based.

Would it be possible to put in some unit tests to ensure the change is working as expected?

File: src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php [hunk 1]
```diff
@@ -535,14 +535,33 @@ public function get_theme_item( $request ) {
...
-		$result   = array(
-			'settings' => $settings,
-			'styles'   => $styles,
+		$theme  = WP_Theme_JSON_Resolver::get_merged_data( 'theme' );
+		$data   = array();
+		$fields = $this->get_fields_for_response( $request );
+
+		if ( rest_is_field_included( 'settings', $fields ) ) {
+			$settings         = $theme->get_settings();
+			$data['settings'] = $settings;
```
Comment (costdev) [inline src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php]:
```suggestion
			$data['settings'] = $theme->get_settings();
```

File: src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php [hunk 2]
```diff
@@ -535,14 +535,33 @@ public function get_theme_item( $request ) {
...
+		}
+
+		if ( rest_is_field_included( 'styles', $fields ) ) {
+			$styles         = $theme->get_raw_data()['styles'];
+			$data['styles'] = $styles;
```
Comment (costdev) [inline src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php]:
```suggestion
			$data['styles'] = $theme->get_raw_data()['styles'];
```

**Review by costdev — CHANGES_REQUESTED** [review]
Just a couple of tidy-up suggestions.

File: src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php [hunk 3]
```diff
@@ -377,12 +377,9 @@ protected function prepare_links( $id ) {
 		$base = sprintf( '%s/%s', $this->namespace, $this->rest_base );
 
 		$links = array(
-			'self'       => array(
+			'self' => array(
 				'href' => rest_url( trailingslashit( $base ) . $id ),
 			),
-			'collection' => array(
```
Comment (spacedmonkey) [inline src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php:383]: There is no collection, so this should never have been here.

**Comment by spacedmonkey** [comment]
Added unit tests.

**Comment by hellofromtonya** [comment]
@spacedmonkey These same changes will be needed in Gutenberg to avoid these changes from being overwritten during a backport update.

**Review by hellofromtonya — APPROVED** [review]
Tested locally and works as expected. Approving for commit 👍

**Review by costdev — APPROVED** [review]

**Comment by spacedmonkey** [comment]
Merged!

**Pull request closed without merging**